index.bs

```diff
@@ -736,6 +736,183 @@ Depending on the underlying platform, the user agent <span class=allow-2119>may<
 
+## Operators ## {#programming-model-operators}
+
+*This section is non-normative.*
+
+The WebNN API defines a set of operators required by well-known CNN and RNN, transformer and generative models that address key [[#usecases-application]]. The details of each operator are defined in the normative sections of this specification, in alphabetical order by the operator name. These operators are grouped into categories based on their functionality in the following non-normative table to give a functional overview of the API surface.
+
+<table id='operators-table' class='data'>
+  <caption>Operators by category</caption>
+  <thead>
+    <tr>
+      <th>Category</th>
+      <th>Operators​</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Tensor creation</td>
+      <td>
+      {{MLGraphBuilder/input()}},
+      {{MLGraphBuilder/constant(tensor)|constant()}}
+      </td>
+    </tr>
+    <tr>
+      <td>Tensor manipulation</td>
+      <td>
+      {{MLGraphBuilder/concat()}},
+      {{MLGraphBuilder/expand()}},
+      {{MLGraphBuilder/gather()}},
+      {{MLGraphBuilder/gatherElements()}},
+      {{MLGraphBuilder/scatterElements()}},
+      {{MLGraphBuilder/gatherND()}},
+      {{MLGraphBuilder/scatterND()}},
+      {{MLGraphBuilder/pad()}},
+      {{MLGraphBuilder/reshape()}},
+      {{MLGraphBuilder/slice()}},
+      {{MLGraphBuilder/split()}},
+      {{MLGraphBuilder/transpose()}},
+      {{MLGraphBuilder/resample2d()}},
+      {{MLGraphBuilder/reverse()}},
+      {{MLGraphBuilder/tile()}}
+      </td>
+    </tr>
+    <tr>
+      <td>Tensor quantization</td>
+      <td>
+      {{MLGraphBuilder/quantizeLinear()}},
+      {{MLGraphBuilder/dequantizeLinear()}}
+      </td>
+    </tr>
+    <tr>
+      <td>Tensor casting</td>
+      <td>
+      {{MLGraphBuilder/cast()}}
+      </td>
+    </tr>
+    <tr>
+      <td>Mathematics</td>
+      <td>
+      {{MLGraphBuilder/add()}},
+      {{MLGraphBuilder/sub()}},
+      {{MLGraphBuilder/mul()}},
+      {{MLGraphBuilder/div()}},
+      {{MLGraphBuilder/max()}},
+      {{MLGraphBuilder/min()}},
+      {{MLGraphBuilder/pow()}},
+      {{MLGraphBuilder/abs()}},
+      {{MLGraphBuilder/ceil()}},
+      {{MLGraphBuilder/cos()}},
+      {{MLGraphBuilder/erf()}},
+      {{MLGraphBuilder/exp()}},
+      {{MLGraphBuilder/floor()}},
+      {{MLGraphBuilder/identity()}},
+      {{MLGraphBuilder/log()}},
+      {{MLGraphBuilder/neg()}},
+      {{MLGraphBuilder/reciprocal()}},
+      {{MLGraphBuilder/sin()}},
+      {{MLGraphBuilder/sqrt()}},
+      {{MLGraphBuilder/tan()}},
+      {{MLGraphBuilder/sign()}}
+      </td>
+    </tr>
+    <tr>
+      <td>Logical</td>
+      <td>
+      {{MLGraphBuilder/equal()}},
+      {{MLGraphBuilder/notEqual()}},
+      {{MLGraphBuilder/greater()}},
+      {{MLGraphBuilder/greaterOrEqual()}},
+      {{MLGraphBuilder/lesser()}},
+      {{MLGraphBuilder/lesserOrEqual()}},
+      {{MLGraphBuilder/logicalNot()}},
+      {{MLGraphBuilder/where()}},
```

Comment:
I suppose the selection operator (aka TOSA select) could be thought of as a logical operator, given it takes boolean input (all the others produce boolean *output*), but I always thought of it as more of a data reorganization operator like `gather`, gathering data respectively from input a or b, depending on the boolean indices (e.g. pseudocode kinda like `gather(concat([a, b], axis:rank), cast(indices, "uint32"), axis:rank)`), and it's not under the [Element-wise logical operations](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-logical) list either.

Reply:
> FYI, CoreML puts select into "control flow" category:

I can see why someone might think that, given it feels like a C ternary operator (a ? b : c), but functionally it's more like a `gatherElements` with 0/1 indices, as it doesn't cause any control flow change in the graph execution (unlike say ONNX `If`/`While`/`Scan` or CoreML `cond`/`while_loop`), and webNN doesn't support any true control flow operators currently, meaning it would be a weirdly isolated category.

Reply:
>  webNN doesn't support any true control flow operators currently, meaning it would be a weirdly isolated category.

That's true.

TOSA places `select` into "Elementwise Ternary Operators" category: https://www.mlplatform.org/tosa/tosa_spec.html#_elementwise_ternary_operators

Should we still have "element-wise binary", "element-wise logical" and "element-wise unary" categories? They are already used in the normative text.

Reply:
This table currently proposes a higher-level functional grouping, and as such abstracts out "element-wise" from "logical" ops and groups both "element-wise binary" and "element-wise unary" ops under "Mathematics".

I'd propose we try to go with a higher-level grouping first, and let the normative text offer more detail where appropriate.